Decred Wallet/Features/TransactionDetails/TransactionDetailsViewController.swift

```diff
@@ -8,262 +8,308 @@ import UIKit
                                                     preferredStyle: UIAlertController.Style.alert)
-            
-            alertController.addAction(UIAlertAction(title: LocalizedStrings.ok,
+
+            let blueTextColorStyle = AttributedStringStyle(tag: "blue",
+                                                           font: UIFont.systemFont(ofSize: 14),
+                                                           color: UIColor.appColors.lightBlue)
+
+            let defaultTextStyle = AttributedStringStyle(font: UIFont.systemFont(ofSize: 14),
+                                                         color: UIColor.appColors.deepGray)
+
+            let infoMessage = Utils.styleAttributedString(LocalizedStrings.tapOnBlueText,
+                                                           styles: [blueTextColorStyle],
+                                                           defaultStyle: defaultTextStyle)
+
+            alertController.setValue(infoMessage, forKey: "attributedMessage")
+            alertController.addAction(UIAlertAction(title: LocalizedStrings.gotIt,
                                                     style: UIAlertAction.Style.default,
                                                     handler: nil))
-            
+
             self.present(alertController, animated: true, completion: nil)
         }
     }
-    
-    func openLink(urlString: String) {
-        if let url = URL(string: urlString) {
-            let viewController = SFSafariViewController(url: url)
-            viewController.delegate = self as SFSafariViewControllerDelegate
-            self.navigationController?.pushViewController(viewController, animated: true)
-        }
+
+    @IBAction func showOrHideDetails(_ sender: Any) {
+        self.isTxDetailsTableViewCollapsed = !self.isTxDetailsTableViewCollapsed
+        self.transactionDetailsTable.reloadData()
+        self.showOrHideDetailsBtn.setTitle(self.isTxDetailsTableViewCollapsed ? LocalizedStrings.showDetails : LocalizedStrings.hideDetails, for: .normal)
     }
 }
 
 extension TransactionDetailsViewController: UITableViewDataSource, UITableViewDelegate {
     func numberOfSections(in tableView: UITableView) -> Int {
-        return 3
+        return self.isTxDetailsTableViewCollapsed ? 1 : 5
     }
-    
+
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return section == 0 ? self.generalTxDetails.count : 1
+        if section == 1 {
+            return self.generalTxDetails.count
+        } else if section == 2 {
+            return self.isTxInputsCollapsed ? 0 : transaction.inputs.count
+        } else if section == 3 {
+            return self.isTxOutputsCollapsed ? 0 : transaction.outputs.count
+        } else {
+            return 1
+        }
     }
-    
+
     func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
-        return 0.5
+        if section == 2 || section == 3 {
```

Comment:
Should probably return height 0 for section 0. The `viewForHeaderInSection` function below returns an empty UIView, suggesting we have nothing to display here. Theoretically speaking (please confirm), if this function returns 0 for a particular section, I don't think a headerview will be requested for that section in the `viewForHeaderInSection` function.

Reply:
You are right.